src/metabase/models/query_field.clj

```diff
@@ -9,47 +9,62 @@
 (doto :model/QueryField
   (derive :metabase/model))
 
-(def ^:private reference-error-joins
-  [[(t2/table-name :model/QueryField) :qf]
-   [:and
-    [:= :qf.card_id :c.id]
-    [:= :qf.explicit_reference true]]
+(defn- coerce-boolean [x]
+  (cond
+    (= 0 x) false
+    (= 0M x) false
+    :else x))
 
-   [(t2/table-name :model/Field) :f]
-   [:and
-    [:= :qf.field_id :f.id]
-    [:= :f.active false]]])
+(defn- coerce-booleans [m]
+  (update-vals m coerce-boolean))
 
 (defn cards-with-reference-errors
   "Given some HoneySQL query map with :model/Card bound as :c, restrict this query to only return cards
-  with invalid references.
-  For now this only handles inactive references, but in future it will also handle unknown references too."
+  with invalid references."
   [card-query-map]
-  ;; NOTE: We anticipate a schema change to support missing references that will result in left-joins and
-  ;; where clauses being appended as well.
-  (update card-query-map :join concat reference-error-joins))
+  (-> card-query-map
+      (update :join concat
+              [[(t2/table-name :model/QueryField) :qf]
+               [:and
+                [:= :qf.card_id :c.id]
+                [:= :qf.explicit_reference true]]])
+      (update :left-join concat
+              [[(t2/table-name :model/Field) :f]
+               [:= :qf.field_id :f.id]])
+      (update :where
+              (fn [existing-where]
+                [:and
+                 existing-where
+                 [:or
+                  [:= :f.id nil]
+                  [:= :f.active false]]]))))
```

Comment:
I think it makes sense to use [honey.sql.helpers](https://cljdoc.org/d/com.github.seancorfield/honeysql/2.6.1147/api/honey.sql.helpers) here, just for the sake of brevity/readability. :)

Reply:
Cool, I'll look into how they can clean it up. Merging as is for now, as the query gets reworked further in the next PR.